pspy/so_spectra.py

```diff
@@ -114,70 +143,54 @@ def bin_spectra(l, cl, binning_file, lmax, type, spectra=None, mbb_inv=None, mcm
     spectra: list of string
       needed for spin0 and spin2 cross correlation, the arrangement of the spectra
     mbb_inv: 2d array
-      the inverse of the binned mode coupling matrix to debiais the spectra
-    mcm_inv: 2d array
-      the inverse of the raw mode coupling matrix to debiais the spectra
-      note that you have to choose between debiasing with the binned or the
-      raw matrix
+      optionnaly apply the inverse of the  mode coupling matrix to debiais the spectra
+    binned_mcm: boolean
+      if mbb_inv is not None, specify if it's binned or not
       
     Return
     ----------
       
     The function return the binned multipole array bin_c and a 1d power spectrum
     array (or dictionnary of 1d power spectra if spectra is not None).
     """
-    
-    #You don't want to deconvolve the binned mcm and the unbinned mcm
-    #You have to choose one of the other
-    #TODO: rework this function logic
-    
-    if mbb_inv is not None and mcm_inv is not None:
-        raise ValueError("You have to choose between binned or raw mcm")
 
-
-    bin_lo, bin_hi, bin_c, bin_size = pspy_utils.read_binning_file(binning_file, lmax)
+    bin_lo, bin_hi, lb, bin_size = pspy_utils.read_binning_file(binning_file, lmax)
     n_bins = len(bin_hi)
+
+    # the alm2cl return cl starting at l = 0, we use spectra from l = 2
+    # this is due in particular to the fact that the mcm is computed only for l>=2
     
-    if type == "Dl":
-        fac = (l * (l + 1) / (2 * np.pi))
-    if type == "Cl":
-        fac = l * 0 + 1
+    if spectra is None: cl = cl[2:lmax]
+    else: cl = {f: cl[f][2:lmax] for f in spectra}
+    
+    l = np.arange(2, lmax)
```

Comment:
I'd put the `l` declaration before everything and then write
```
if spectra is None: cl = cl[l]
else: cl = {f: cl[f][l] for f in spectra}
```
Maybe `ell` is a better name.